### PR TITLE
 check table is empty before do import

### PR DESCRIPTION
**Pull request by july2993**
### What problem does this PR solve? 
if the user DO NOT know lighting can only init an empty table, it may
corrupt the table, also, will fail to pass checksum after import.

### What is changed and how it works?
check table is empty before do import data.
if it is not empty, print the related error to force the user to clean up the table if the user wants to import the table.

### Check List 

Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
  
if the table is empty run success, if the table is not empty runs fail like this:
```
[2021/03/30 17:07:31.064 +08:00] [INFO] [restore.go:612] ["restore all schema completed"] [takeTime=15.574742ms] []
[2021/03/30 17:07:31.066 +08:00] [ERROR] [restore.go:327] ["run failed"] [step=2] [error="table `test`.`test1` not empty, please cleanup the table first"]
Error: table `test`.`test1` not empty, please cleanup the table first
[2021/03/30 17:07:31.066 +08:00] [ERROR] [restore.go:338] ["the whole procedure failed"] [takeTime=22.20738ms] [error="table `test`.`test1` not empty, please cleanup the table first"]
[2021/03/30 17:07:31.066 +08:00] [ERROR] [main.go:91] ["tidb lightning encountered error stack info"] [error="table `test`.`test1` not empty, please cleanup the table first"] [errorVerbose="table `test`.`test1` not empty, please cleanup the table first\ngithub.com/pingcap/br/pkg/lightning/restore.(*RestoreController).checkTableEmpty\n\tgithub.com/pingcap/br/pkg/lightning/restore/restore.go:591\ngithub.com/pingcap/br/pkg/lightning/restore.(*RestoreController).restoreSchema\n\tgithub.com/pingcap/br/pkg/lightning/restore/restore.go:632\ngithub.com/pingcap/br/pkg/lightning/restore.(*RestoreController).Run\n\tgithub.com/pingcap/br/pkg/lightning/restore/restore.go:314\ngithub.com/pingcap/br/pkg/lightning.(*Lightning).run\n\tgithub.com/pingcap/br/pkg/lightning/lightning.go:313\ngithub.com/pingcap/br/pkg/lightning.(*Lightning).RunOnce\n\tgithub.com/pingcap/br/pkg/lightning/lightning.go:191\nmain.main.func2\n\tcommand-line-arguments/main.go:86\nmain.main\n\tcommand-line-arguments/main.go:88\nruntime.main\n\truntime/proc.go:203\nruntime.goexit\n\truntime/asm_amd64.s:1373"]
[2021/03/30 17:07:31.066 +08:00] [ERROR] [main.go:92] ["tidb lightning encountered error"] [error="table `test`.`test1` not empty, please cleanup the table first"]
tidb lightning encountered error:  table `test`.`test1` not empty, please cleanup the table first
```
 - No code

Code changes

Side effects

 

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation

### Release Note

 - No release note

### CODE REVIEW

**Comment by glorv** [comment]
@IANTHEREAL PTAL

I think this PR is conflict with #784.

**Comment by july2993** [comment]
> @IANTHEREAL PTAL
> 
> I think this PR is conflict with #784.

does it assume the data being imported has no conflict with the existing data(different pk/uk) in #784 ?

File: pkg/lightning/restore/restore.go [hunk 1]
```diff
@@ -607,6 +628,19 @@ func (rc *RestoreController) restoreSchema(ctx context.Context) error {
 	}
 	rc.dbInfos = dbInfos
 
+	if !rc.cfg.Checkpoint.Enable &&
+		(rc.cfg.TikvImporter.Backend == config.BackendLocal || rc.cfg.TikvImporter.Backend == config.BackendImporter) {
```
Comment (kennytm) [inline pkg/lightning/restore/restore.go:632]:
```suggestion
		rc.cfg.TikvImporter.Backend != config.BackendTiDB {
```

**Comment by july2993** [comment]
@kennytm @glorv PTAL

**Pull request merged**